### PR TITLE
conflicting option names

### PR DESCRIPTION
**Pull request by phlptp**
Take the configurability of an option into account when determining ambiguous names and conflicts.

### CODE REVIEW

**Comment by phlptp** [comment]
This will fix #1047 
This will resolve #1048

**Comment by phlptp** [comment]
I made the changes in  PR #1049.   There were a few edge cases to work through that the fuzzer caught.  
There is probably a bit more to do to take into account whether the subcommand or app itself is not configurable.   That might require some additional thought.  But for now this will resolve the `--h` issue and any other case where one of the options is not configurable, and hence wouldn't present an ambiguity in the config files.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -87,7 +87,7 @@ repos:
         exclude: .pre-commit-config.yaml
 
   - repo: https://github.com/codespell-project/codespell
-    rev: v2.2.6
+    rev: v2.3.0
     hooks:
       - id: codespell
-        args: ["-L", "atleast,ans,doub,inout"]
+        args: ["-L", "atleast,ans,doub,inout,AtMost"]
```
Comment (henryiii) [inline .pre-commit-config.yaml:93]: Oh, did this bug get fixed in 2.3.0? It used to be you had to put all lower case versions here.

File: tests/fuzzFail/fuzz_app_file_fail40 [hunk 2]
```diff
@@ -0,0 +1 @@
+     config                      ' '
```
Comment (henryiii) [inline tests/fuzzFail/fuzz_app_file_fail40:1]: Haven't followed the fuzzer that closely, what does this do?

**Review by henryiii — APPROVED** [review]

File: tests/fuzzFail/fuzz_app_file_fail40 (hunk 2 above)
Comment (phlptp) [reply tests/fuzzFail/fuzz_app_file_fail40:1]: Right now the fuzzer has a large number of options,  and some limited mechanics to generate new ones.  It processes the options making sure there is nothing that generates a seg fault or other bad condition.   Then generates a config file from the results and reads that config file back in.      This led to all kinds of inconsistencies that got fixed a few months ago.   At some point I will make it so it checks to make sure the values read from the config file are actually the same as the original values, then I might consider submitting it to [OSS-Fuzz](https://github.com/google/oss-fuzz).   The current test just runs a quick_fuzz ~6 minutes and can catch some issues especially when changing some internal mechanics or other things.    I will keep adding to it as I have time.  

This particular test caught an issue with creating a positional option named config, which was newly allowed.  This then triggered an error as it was trying to match with the config option which was not configurable, so had to make it so it checked for other options if some were not configurable.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (phlptp) [reply .pre-commit-config.yaml:93]: Didn't know there was a bug, but this resolved the pre-commit issue

**Pull request merged**